File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.js
var areAlmostEqual = function(s1, s2) {
    let diff = [];

    for (let i = 0; i < s1.length; i++) {
        if (s1[i] !== s2[i]) {
            diff.push(i);
        }
    }

    if (diff.length === 0) return true;
    if (diff.length !== 2) return false;

    let a = diff[0];
    let b = diff[1];

    return s1[a] === s2[b] && s1[b] === s2[a];
};

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1,s2):
        a=[]
        for i in range(len(s1)):
            if s1[i]!=s2[i]:
                a.append(i)
        if len(a)==0:
            return True
        if len(a)!=2:
            return False
        i,j=a
        return s1[i]==s2[j] and s1[j]==s2[i]